src/gen/luacats_parser.lua

```diff
@@ -224,6 +226,11 @@ local function process_doc_line(line, state)
   elseif kind == 'enum' then
     -- TODO
     state.doc_lines = nil
+  elseif kind == 'async' then
+    cur_obj.async = true
```

Comment:
Is this a `@async` tag? Can you say a bit about the idea there, why is it needed, how will these functions differ vs regular functions?

Reply:
The `@async` tag is used by both emmylua and luals (which we disable it atm), it basically indicates that a function has a code path to a `coroutine.yield` call.

In `vim.async` speak, `vim.async.run()` only accepts `async fun()` (a function with the `@async` tag) as an input, and `async.wrap()` inputs a `function` and outputs an `async function`.

```lua
--- @async
local function foo()
   -- @async needed above because this functions calls async.await
   async.await(...)
end

async.run(foo) -- all good as foo is async

foo() -- error cannot call async function in non-async context.
```

Reply:
> In `vim.async` speak, `vim.async.run()` only accepts `async fun()` (a function with the `@async` tag) as an input, and `async.wrap()` inputs a `function` and outputs an `async function`.

can we just always `wrap()` instead of making the caller care ?

is this comparable to the "what color is your function" criticism of async/await ? https://news.ycombinator.com/item?id=41898204

Reply:
> You still can’t use them with exception handling or other control flow statements. You still can’t call a function that returns a future from synchronous code. (Well, you can, but if you do, the person who later maintains your code will invent a time machine, travel back in time to the moment that you did this and stab you in the face with a #2 pencil.)

Neither of these points apply here. 

`pcall` works as normal.

You can call an async function from a synchronous context, you've just got to `async.run()` it and then wait synchronously for it to complete.